content/assignments/hello-world/auto-grading-test.md

```diff
@@ -0,0 +1,87 @@
+Use the tests to check your work as you complete the assignment. The goal is to pass all tests.
+
+Let’s learn how to run the tests in Visual Studio.
+
+1. Each assignment will contain a test **project**. This project contains all of the auto-grading tests and may have more than one test class. In the Solution/Solution Explorer, navigate to `HelloTests.cs` inside the `Hello.Tests` project. 
+
+   {{% notice blue "Note" "rocket" %}}
+   You will not need to open the `Using.cs` file.  This file connects the testing project to the testing library.  This is not a test class.
+   {{% /notice %}}
+
+   A **test** is a single method in a test file with `[TestMethod]` above it. 
+
+      ```csharp
+      [TestClass]
+      public class AutogradingTests
+      {
+         [TestMethod]
+         public void TestToDemonstrateWhatATestMethodLooksLike()
+         {
+            //test code here
+         }
+      }
+      ```
+
+      A test class may contain many tests. 
+
+
+      ```csharp
+      [TestClass]
+      public class AutogradingTests
+      {
+         [TestMethod]
+         public void TestToDemonstrateWhatATestMethodLooksLike()
+         {
+            //test code here
+         }
+
+         [TestMethod]
+         public void SecondTestToDemonstrateWhatATestMethodLooksLike()
+         {
+            //test code here
+         }
+      }
+   ```
+1. Run all of your tests using the Test Explorer window.
+   1. Mac Users: 
+      1. **View** > **Tests**  
+      1. Select the green triangle on the left of the Test Explorer Window. 
+      1. In the Test Explorer, step through the file tree until you find the failed test. 
+      1. If you need more help running the tests, checkout this [guide](https://learn.microsoft.com/en-us/visualstudio/mac/testing?view=vsmac-2022#running-tests) for running tests in Visual Studio for Mac
+   1. Windows Users 
+      1. **Test** > **Windows** > **Test Explorer**
+      1. Select the `Run All` icon to run all of your tests. 
+      1. In the Test Explorer, step through your file tree until you find the failed test.
+      1. If you need more help running the tests, checkout this [guide](https://learn.microsoft.com/en-us/visualstudio/test/run-unit-tests-with-test-explorer?view=vs-2022) for running tests in Visual Studio.
+
+
+1. Visual Studio opens a Test Results Pane every time it runs a test. This pane provides you with detailed information about which tests passed and which tests failed.  The pane will also provide you with output for any failed tests, including what was expected and what was actually received.  This is a great place to look when dealing with a failing test.
+
+   Now that you have run the tests, you should see that one of the tests has failed and the other passed. We have not touched any code yet. Use the Test Results Pane to navigate into which specific test has failed.
```

Comment:
```suggestion
   Now that you have run the tests, you should see that one of the tests has failed and the other passed. We have not touched any code yet. Using the Test Results Pane double click on the test name to navigate to the test code that has failed.
```

Reply:
This sentence just seems off.  What are we suppose to be looking at?  There is a little carrot I can click that expands and gives me information on why the test failed but I also get that to the right of the tests as well.

Reply:
in this example, we only have 1 test class with 2 tests. in A2 and A4 you may have up to 8 tests in a test class.  those assignments also have multiple test classes.  there is a lot more scrolling to be done.  while it might seem obvious, i like the word "navigate" because the tests and test classes will get more complex as the assignments grow.